tpm/src/socket.rs

```diff
@@ -0,0 +1,166 @@
+// Copyright © 2022, Microsoft Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+//
+
+use crate::ioctl::TPM_CRB_BUFFER_MAX;
+use anyhow::anyhow;
+use std::io::Read;
+use std::os::unix::io::{AsRawFd, RawFd};
+use std::os::unix::net::UnixStream;
+use std::thread::sleep;
+use std::time::{Duration, Instant};
+use thiserror::Error;
+use vmm_sys_util::sock_ctrl_msg::ScmSocket;
+
+#[derive(Error, Debug)]
+pub enum TPMSocError {
+    #[error("Cannot connect to TPM Socket even after retrying for 10 secs")]
+    ConnectToTPMSocket(#[source] anyhow::Error),
```

Comment:
You don't need to add the `TPM` here; it's self evident from the context.

Reply:
fixed